### ISSUE TITLE
Error when loading devtools

### ISSUE THREAD

**Issue by shirdekel** [r-lib/devtools#2309]
I get the following error when loading `devtools`. Hasn't happened before, and I'm not sure how to debug it further.

``` r
require(devtools)
#> Loading required package: devtools
#> Loading required package: usethis
#> Error in get(genname, envir = envir) : object 'testthat_print' not found
```

<sup>Created on 2021-01-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       macOS Mojave 10.14.6        
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Sydney            
#>  date     2021-01-18                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date       lib source        
#>  assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)
#>  callr         3.4.4   2020-09-07 [1] CRAN (R 4.0.2)
#>  cli           2.2.0   2020-11-20 [1] CRAN (R 4.0.2)
#>  crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)
#>  desc          1.2.0   2018-05-01 [1] CRAN (R 4.0.0)
#>  devtools    * 2.3.2   2020-09-18 [1] CRAN (R 4.0.2)
#>  digest        0.6.27  2020-10-24 [1] CRAN (R 4.0.2)
#>  ellipsis      0.3.1   2020-05-15 [1] CRAN (R 4.0.0)
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.0.0)
#>  fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)
#>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.0.2)
#>  glue          1.4.2   2020-08-27 [1] CRAN (R 4.0.2)
#>  highr         0.8     2019-03-20 [1] CRAN (R 4.0.0)
#>  htmltools     0.5.1   2021-01-12 [1] CRAN (R 4.0.2)
#>  knitr         1.30    2020-09-22 [1] CRAN (R 4.0.2)
#>  lifecycle     0.2.0   2020-03-06 [1] CRAN (R 4.0.0)
#>  magrittr      2.0.1   2020-11-17 [1] CRAN (R 4.0.2)
#>  memoise       1.1.0   2017-04-21 [1] CRAN (R 4.0.0)
#>  pkgbuild      1.1.0   2020-07-13 [1] CRAN (R 4.0.2)
#>  pkgload       1.1.0   2020-05-29 [1] CRAN (R 4.0.0)
#>  prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.0.0)
#>  processx      3.4.4   2020-09-03 [1] CRAN (R 4.0.2)
#>  ps            1.3.4   2020-08-11 [1] CRAN (R 4.0.2)
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.0.0)
#>  R6            2.5.0   2020-10-28 [1] CRAN (R 4.0.2)
#>  remotes       2.2.0   2020-07-21 [1] CRAN (R 4.0.2)
#>  rlang         0.4.10  2020-12-30 [1] CRAN (R 4.0.2)
#>  rmarkdown     2.5     2020-10-21 [1] CRAN (R 4.0.2)
#>  rprojroot     2.0.2   2020-11-15 [1] CRAN (R 4.0.2)
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)
#>  stringi       1.5.3   2020-09-09 [1] CRAN (R 4.0.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.0.0)
#>  testthat      2.3.2   2020-03-02 [1] CRAN (R 4.0.0)
#>  usethis     * 2.0.0   2020-12-10 [1] CRAN (R 4.0.2)
#>  withr         2.3.0   2020-09-22 [1] CRAN (R 4.0.2)
#>  xfun          0.19    2020-10-30 [1] CRAN (R 4.0.2)
#>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.0.0)
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by ttimbers** [comment]
I observed this as well! It was when I loaded the `testthat` library. The environment I observed it in is captured in this Docker image - it is very strange! This was with `testthat` version 2.3.2 and R version 4.0.3.

- [Docker image](https://hub.docker.com/layers/132772151/ubcdsci/r-dsci-100/v0.13.0/images/sha256-2c6fe469d47138705fddc0670b592e4c7f054bed8391d486c032def0ba31a6ca?context=explore)
- [Dockerfile]( https://github.com/UBC-DSCI/docker-stacks/blob/v0.13.0/r-dsci-100/Dockerfile)

Downgrading R to version 4.0.0 in another version of this Docker image resolved this (of course other things might have other things that differed between these two Docker images as I did not pin the versions of all the things). Here are the details for this one. The `testthat` version was still held at 2.3.2 here.

- [Docker image](https://hub.docker.com/layers/ubcdsci/r-dsci-100/v0.15.0/images/sha256-7ee5a6ed14a55a204375ce2252d19f29c2bd0f7acd8323ff7b202badce19ba73?context=explore)
- [Dockerfile]( https://github.com/UBC-DSCI/docker-stacks/blob/v0.15.0/r-dsci-100/Dockerfile)

**Comment by shirdekel** [comment]
My guess is that it's something to do with the package version, because I do not get this error in my other machine, and they have the same R version:

    require(devtools)
    #> Loading required package: devtools
    #> Loading required package: usethis

<sup>Created on 2021-01-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
<summary>
Session info
</summary>

    devtools::session_info()
    #> ─ Session info ───────────────────────────────────────────────────────────────
    #>  setting  value                       
    #>  version  R version 4.0.2 (2020-06-22)
    #>  os       macOS Mojave 10.14.6        
    #>  system   x86_64, darwin17.0          
    #>  ui       X11                         
    #>  language (EN)                        
    #>  collate  en_AU.UTF-8                 
    #>  ctype    en_AU.UTF-8                 
    #>  tz       Australia/Sydney            
    #>  date     2021-01-18                  
    #> 
    #> ─ Packages ───────────────────────────────────────────────────────────────────
    #>  package     * version date       lib source        
    #>  assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)
    #>  callr         3.5.1   2020-10-13 [1] CRAN (R 4.0.2)
    #>  cli           2.2.0   2020-11-20 [1] CRAN (R 4.0.2)
    #>  crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)
    #>  desc          1.2.0   2018-05-01 [1] CRAN (R 4.0.0)
    #>  devtools    * 2.3.0   2020-04-10 [1] CRAN (R 4.0.0)
    #>  digest        0.6.27  2020-10-24 [1] CRAN (R 4.0.2)
    #>  ellipsis      0.3.1   2020-05-15 [1] CRAN (R 4.0.0)
    #>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.0.0)
    #>  fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)
    #>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.0.2)
    #>  glue          1.4.2   2020-08-27 [1] CRAN (R 4.0.2)
    #>  highr         0.8     2019-03-20 [1] CRAN (R 4.0.0)
    #>  htmltools     0.5.0   2020-06-16 [1] CRAN (R 4.0.2)
    #>  knitr         1.30    2020-09-22 [1] CRAN (R 4.0.2)
    #>  magrittr      2.0.1   2020-11-17 [1] CRAN (R 4.0.2)
    #>  memoise       1.1.0   2017-04-21 [1] CRAN (R 4.0.0)
    #>  pkgbuild      1.2.0   2020-12-15 [1] CRAN (R 4.0.2)
    #>  pkgload       1.1.0   2020-05-29 [1] CRAN (R 4.0.0)
    #>  prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.0.0)
    #>  processx      3.4.5   2020-11-30 [1] CRAN (R 4.0.2)
    #>  ps            1.5.0   2020-12-05 [1] CRAN (R 4.0.2)
    #>  R6            2.5.0   2020-10-28 [1] CRAN (R 4.0.2)
    #>  remotes       2.2.0   2020-07-21 [1] CRAN (R 4.0.2)
    #>  rlang         0.4.10  2020-12-30 [1] CRAN (R 4.0.2)
    #>  rmarkdown     2.5     2020-10-21 [1] CRAN (R 4.0.2)
    #>  rprojroot     2.0.2   2020-11-15 [1] CRAN (R 4.0.2)
    #>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)
    #>  stringi       1.5.3   2020-09-09 [1] CRAN (R 4.0.2)
    #>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.0.0)
    #>  testthat      3.0.1   2020-12-17 [1] CRAN (R 4.0.2)
    #>  usethis     * 1.6.3   2020-09-17 [1] CRAN (R 4.0.2)
    #>  withr         2.3.0   2020-09-22 [1] CRAN (R 4.0.2)
    #>  xfun          0.19    2020-10-30 [1] CRAN (R 4.0.2)
    #>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.0.0)
    #> 
    #> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library

</details>

**Comment by jimhester** [comment]
This is likely due to changes in the usethis and testthat packages. Nothing has changed in the devtools package recently. I would suggest you use the latest versions of testthat and usethis.

**Comment by shirdekel** [comment]
This was fixed for me after updating `testthat`.

**Comment by r2evans** [comment]
For more info (no need to reopen): this was caused by an upgrade to `rlang` that did not trigger an update of `testthat`.

Specifically, `testthat::testthat_print` was added (https://github.com/r-lib/testthat/commit/07b19c87e2a3465ceb155570bd86ae68c75a2350) before testthat3 was released. `rlang` started using it around the same time (https://github.com/r-lib/rlang/commit/bfb1d4130a811a99ca190cc88077eab08ea2a5d8), before it released rlang-0.4.10. So far, so good.

Unfortunately, `testthat` is in rlang's `Suggests:`, and the default for `install.packages` is `dependencies=NA` which only considers `c("Depends", "Imports", "LinkingTo")`.
